### DISCUSSION TITLE
How to update IUserDefinition defination in serenity?? I want to add UserIDStr field to it as I have two Ids UserId and UserIDStr. Need help on urgent basis

### DISCUSSION THREAD

**Discussion by MinalRaorane** [serenity-is/Serenity#7163] [Help]
How to update IUserDefinition defination in serenity?? I want to add UserIDStr field to it as I have two Ids UserId and UserIDStr. Need help on urgent basis

**Comment by VictorTomaili** [comment]
https://github.com/serenity-is/serene/blob/master/src/Serene.Web/Modules/Administration/User/Authentication/UserRetrieveService.cs#L35

**Reply by MinalRaorane** [reply]
@VictorTomaili Will you please more elaborate on how to solve this problem?? As I have already addded UserIDStr in UserDefination but it is not reflecting in IUserDefinition as follows:
![image](https://github.com/serenity-is/Serenity/assets/144375037/6c8ed850-4e6f-49dc-aa98-7922d470cb3a)
![image](https://github.com/serenity-is/Serenity/assets/144375037/8ee6dcf7-a2db-4119-8e93-c9ab06a6b3c8)
![image](https://github.com/serenity-is/Serenity/assets/144375037/d4933ba3-c850-4807-bd72-7d5ca70746ab)

currentuser showing only Id, UserName, DisplayName, Email, IsActive. as IUserdefination have defined only these 5 fields I want to add one more in this i..e UserIdStr. Please help

**Reply by MinalRaorane** [reply]
I also want to get this UserIDStr in DashboardPage.cs in Index Method. Really need help.

**Reply by VictorTomaili** [reply]
`var user = (UserDefinition)userRetrieveService.ById(userId);`

**Reply by NikolaS4yourb** [reply]
Hi, 
can you please give us solution for dialog ? I'm also  unable to see UserId, or any other added field.
Tnx

**Reply by VictorTomaili** [reply]
https://github.com/serenity-is/serene/blob/8362132e9f039393af67dd011a224326f2fecec9/src/Serene.Web/Modules/Administration/User/Authentication/PermissionService.cs#L53
